### ISSUE TITLE
RMG cannot finds reactions whose inverse reactions generating CO

### ISSUE THREAD

**Issue by xiaoruiDong** [ReactionMechanismGenerator/RMG-Py#1345]
Recently, I am dealing with the following reaction 

![image](https://user-images.githubusercontent.com/35771233/38956950-25a992f4-4327-11e8-87da-8087f9541a9d.png)

and no kinetics data has been found for this reaction. However its reverse reaction, belonging to the family 'R_Addition_COm', could be found. Other than this specific reaction and family, I also found that if the reaction does not match a library or training reaction, the reverse reaction from family 'R_Addition_COm', 'R_Addition_CSm', '1,2_Insertion_CO', '2+2 cycloaddition_CO', 'CO_Disproportionation' will not be found or matched to kinetics data. 

According to my discussion with Max and Alon, this may relates to the generation of reverse templates, but still need to be inspected.

**Comment by mliu49** [comment]
From our debugging this morning, I can pretty confidently say that it's due to generation of product templates, at least in the case of `R_Addition_COm`.

The two reactant templates are:
```
# COm
1 *1 C2tc u0 p1 c-1 {2,T}
2 *3 O4tc u0 p1 c+1 {1,T}

# Y_rad
1 *2 R u1
```

And the automatically generated product template is:
```
# YC.=O
1 *1 [Cdc,Cd,CS,CO] u1 p0 c-1 {2,D} {3,S}
2 *3 [O2s,O2sc]     u0 p2 c+1 {1,D}
3 *2 R              u0 {1,S}
```

During product template generation, we apply the recipe to the reactant templates but don't do any updating of the resulting Group (see family.py line 1300). Thus, all attributes are unchanged unless specified by the reaction recipe.

I think there are two potential solutions: 1) we add charge modification to reaction recipes, or 2) we add a Group.update() method that can somehow identify how charges should change.

**Comment by alongd** [comment]
Could tell RMG to use a product template, if available, otherwise generate it?

**Comment by mliu49** [comment]
Addressed by #1356 and ReactionMechanismGenerator/RMG-database#268.
